packages/edit-site/src/components/header/document-actions/index.js

```diff
@@ -71,11 +79,27 @@ function useSecondaryText() {
 	return {};
 }
 
-export default function DocumentActions( { documentTitle } ) {
+export default function DocumentActions( { template } ) {
+	const documentTitle = template?.slug;
 	const { label, isActive } = useSecondaryText();
+
 	// Title is active when there is no secondary item, or when the secondary
 	// item is inactive.
 	const isTitleActive = ! label?.length || ! isActive;
+
+	// The anchor rect is used to position the dropdown relative to the overall
+	// main title area, not to the chevron button. We use a memo to make sure
+	// the props object reference is the same to avoid re-renders. NOTE: A
+	// limitation of `anchorRect` is that it does not move when resizing.
+	const titleRef = useRef( null );
+	const popoverProps = useMemo(
+		() =>
+			titleRef.current
+				? { anchorRect: titleRef.current.getBoundingClientRect() }
```

Comment:
I'm unsure how to fix the issue where this will not update as the window resizes. For example, when the nav sidebar opens, the values of the `anchorRect` will change until that is finished. Same for resizing the window.

Even if we pass this value directly without using a memo, it will only be recomputed as often as the component re-renders. It does not currently re-render often enough to update the `anchorRect` after opening the nav sidebar. 

I wonder how the popover component handles this with the basic ref to the button. It does update the position correctly in that case.

Reply:
I've solved this by passing the ref directly.